version: '3'
services:
  db:
    image: postgres:16-alpine
    restart: on_failure
    environment:
      POSTGRES_DB: drfhw
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: Msssvintsova006
    volumes:
      - pg_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER}"]
      interval: 10s
      retries: 5
      timeout: 5s

  app:
    build: .
    tty: true
    ports:
      - "8000:8000"
    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - .:/app

volumes:
  pg_data:
